client-app/src/Components/DonorForm.tsx

```diff
@@ -99,65 +114,169 @@ const DonorForm: React.FC = () => {
+              onChange={handleChange}
+              className={`w-full px-3 py-2 rounded border ${errors.addressLine2 ? 'border-red-500' : 'border-gray-300'}`}
+            />
             {errors.addressLine2 && <p className="text-red-500 text-sm mt-1">{errors.addressLine2}</p>}
           </div>
-          <div className="mb-4">
-            <label htmlFor="state" className="block text-sm font-semibold mb-1">State:</label>
-            <input type="text" id="state" name="state" value={formData.state} onChange={handleChange} className={`w-full px-3 py-2 rounded border ${errors.state ? 'border-red-500' : 'border-gray-300'}`} />
+
+          <div className="form-field">
+            <label htmlFor="state" className="block text-sm font-semibold mb-1">
+              State <span className="text-red-500">*</span>
+            </label>
+            <input
+              type="text"
+              id="state"
+              name="state"
+              value={formData.state}
+              onChange={handleChange}
+              className={`w-full px-3 py-2 rounded border ${errors.state ? 'border-red-500' : 'border-gray-300'}`}
+            />
             {errors.state && <p className="text-red-500 text-sm mt-1">{errors.state}</p>}
           </div>
-          <div className="mb-4">
-            <label htmlFor="city" className="block text-sm font-semibold mb-1">City:</label>
-            <input type="text" id="city" name="city" value={formData.city} onChange={handleChange} className={`w-full px-3 py-2 rounded border ${errors.city ? 'border-red-500' : 'border-gray-300'}`} />
+
+          <div className="form-field">
+            <label htmlFor="city" className="block text-sm font-semibold mb-1">
+              City <span className="text-red-500">*</span>
+            </label>
+            <input
+              type="text"
+              id="city"
+              name="city"
+              value={formData.city}
+              onChange={handleChange}
+              className={`w-full px-3 py-2 rounded border ${errors.city ? 'border-red-500' : 'border-gray-300'}`}
+            />
             {errors.city && <p className="text-red-500 text-sm mt-1">{errors.city}</p>}
           </div>
-          <div className="mb-4">
-            <label htmlFor="zipcode" className="block text-sm font-semibold mb-1">Zip Code:</label>
-            <input type="text" id="zipcode" name="zipcode" value={formData.zipcode} onChange={handleChange} className={`w-full px-3 py-2 rounded border ${errors.zipcode ? 'border-red-500' : 'border-gray-300'}`} />
+
+          <div className="form-field">
+            <label htmlFor="zipcode" className="block text-sm font-semibold mb-1">
+              Zip Code <span className="text-red-500">*</span>
+            </label>
+            <input
+              type="text"
+              id="zipcode"
+              name="zipcode"
+              value={formData.zipcode}
+              onChange={handleChange}
```

Comment:
I'm just starting to learn HTML so I may be incorrect, but I feel like there might be a more effective way to create these fields, since it looks like first name, last name, contact, email, address lines, state, city, and zip code all have very similar formatting and code.

Reply:
Feel free to use AI to get suggestions for improving the code